addons/web_editor/static/src/components/html_history_dialog/html_history_dialog.js

```diff
@@ -0,0 +1,124 @@
+/** @odoo-module **/
+
+import { Dialog } from '@web/core/dialog/dialog';
+import { useService } from "@web/core/utils/hooks";
+import {Component, onMounted, useState, markup} from '@odoo/owl';
+import { _t } from 'web.core';
+const { DateTime } = luxon;
+
+class HtmlHistoryDialog extends Component {
+
+    state = useState({
+        latestHistoryDiffs: [],
+        currentHtml: null,
+        currentComparisonHtml: null,
+        currentId: null,
+        buttonDisabled: true,
+    });
+    setup() {
+        this.size = 'xl';
+        this.title = _t("Recent History");
+        this.orm = useService("orm");
+
+        this.comparisonCache = {};
+        this.htmlCache = {};
+
+        onMounted(() => this.init());
+    }
+
+    async init(limit=100) {
+        // limit the number of record and order them by id desc
+        const recIds = this.props.recordsIds.slice(0, limit).reverse();
+
+        // get record data
+        this.state.latestHistoryDiffs = await this.orm.call(
+            'field.html.history.diff',
+            'read',
+            [recIds, ["id", "create_date", "create_uid"]]
+        );
+
+        // get the comparison html for the latest diff
+        this.state.buttonDisabled = false;
+        await this.updateCurrentComparisonHtml(this.state.latestHistoryDiffs[0].id);
+    }
+
+    async updateCurrentComparisonHtml(historyDiffId) {
```

Comment:
I'm not really fond of the "comparison html" monicker. Personally I'd just call it something like "updateCurrentCompare", but I suppose it's maybe a tad too vague. I think an easy way to change the name so that it doesn't tickle me and stays close to the original intention would be `updateCurrentHtmlComparison`.

Reply:
sure

Reply:
It has been refactored as `updateCurrentRevision(revisionId) ` because both the revision **content** and the **comparison** of the current content to the said revision are updated.

Tell me if it's ok for you.